packages/rest/test/unit/deserializer.test.ts

```diff
@@ -0,0 +1,193 @@
+// Copyright IBM Corp. 2017,2018. All Rights Reserved.
+// Node module: @loopback/rest
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {deserialize} from '../..';
+import {expect, ShotRequest, ShotRequestOptions} from '@loopback/testlab';
+import {OperationObject, ParameterObject} from '@loopback/openapi-spec';
+
+// tslint:disable:no-any
+describe('deserializer', () => {
+  function expectToDeserialize(
+    param: ParameterObject,
+    source: any[],
+    target: any[],
+  ) {
+    expect(source.map(i => deserialize(i, param))).to.eql(target);
+  }
+
+  function expectToDeserializeNullOrUndefined(param: ParameterObject) {
+    expect(deserialize(null, param)).to.be.null();
+    expect(deserialize(undefined, param)).to.be.undefined();
+  }
+
+  function expectToFail(
```

Comment:
Please move all helper function to the bottom of the test file, see http://loopback.io/doc/en/contrib/style-guide.html#layout-of-test-files (the section addresses hook function, not helper functions, but the principle is the same).